syntax/scan.go

```diff
@@ -196,29 +196,39 @@ func (p Position) String() string {
 	return fmt.Sprintf("%s:%d:%d", p.Filename(), p.Line, p.Col)
 }
 
+func (p Position) isBefore(q Position) bool {
+	if p.Line != q.Line {
+		return p.Line < q.Line
+	}
+	return p.Col < q.Col
+}
+
 // An scanner represents a single input file being parsed.
 type scanner struct {
-	complete  []byte   // entire input
-	rest      []byte   // rest of input
-	token     []byte   // token being scanned
-	pos       Position // current input position
-	depth     int      // nesting of [ ] { } ( )
-	indentstk []int    // stack of indentation levels
-	dents     int      // number of saved INDENT (>0) or OUTDENT (<0) tokens to return
-	lineStart bool     // after NEWLINE; convert spaces to indentation tokens
+	complete     []byte    // entire input
+	rest         []byte    // rest of input
+	token        []byte    // token being scanned
+	pos          Position  // current input position
+	depth        int       // nesting of [ ] { } ( )
+	indentstk    []int     // stack of indentation levels
+	dents        int       // number of saved INDENT (>0) or OUTDENT (<0) tokens to return
+	lineStart    bool      // after NEWLINE; convert spaces to indentation tokens
+	keepComments bool      // if false, do not return comments tokens
```

Comment:
This comment is stale. Also, document a boolean using a verb phrase:

// accumulate comments in slice

Reply:
Done